### DISCUSSION TITLE
Handle multiple databases

### DISCUSSION THREAD

**Discussion by michele-grifa** [jowilf/starlette-admin#407] [Q&A]
I have a FastAPI project that involve multiple databases, and I'm finding any information on how to handle it the administrative interface

**Comment by jowilf** [comment]
I think the easiest way to handle multiple database backends per session right now is to override the [Admin](https://github.com/jowilf/starlette-admin/blob/dcf8aa8439c4463edee6587d3a66a5e160c9e8ae/starlette_admin/contrib/sqla/admin.py#L24) class and provide your own [DBSessionMiddleware](https://github.com/jowilf/starlette-admin/blob/dcf8aa8439c4463edee6587d3a66a5e160c9e8ae/starlette_admin/contrib/sqla/admin.py#L56). More information is available on the [SQLAlchemy documentation](https://docs.sqlalchemy.org/en/20/orm/persistence_techniques.html#session-partitioning)

we can extend the Admin class to support also a [sessionmaker](https://docs.sqlalchemy.org/en/20/orm/session_basics.html#using-a-sessionmaker). That will make it easier in the future
